### PR TITLE
Upgrade typescript & other packages

### LINKED ISSUE #147: Upgrade to latest typescript
Upgrade to latest typescript and other tools (mocha, chai, etc...). 

```
npm install typescript@latest
```

Here are some good candidates:
 - typescript
 - chai
 - coveralls-next
 - eslint
 - mocha
 - nyc
 - q
 - rimraf
 - sinon
 - source-map-support
 - ts-node

### PR DESCRIPTION
**Pull request by MilapNaik**
Fixes #147

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,15 +52,15 @@
...
+        "eslint": "^8.57.0",
+        "mocha": "^10.3.0",
         "nyc": "^15.1.0",
         "q": "^1.5.1",
-        "rimraf": "^2.6.2",
-        "sinon": "^11.1.2",
-        "source-map-support": "^0.5.13",
-        "ts-node": "^10.3.1",
-        "typescript": "^4.4.4"
+        "rimraf": "^5.0.5",
```
Comment (TwitchBronBron) [inline package.json:59]: Looks like rimraf is failing the build. Perhaps we need to roll back to rimraf v4?

Comment (MilapNaik) [reply package.json:59]: Actually I had to rollback rimraf to v3 AND typescript to v4 in order to stop getting errors with node 12.22.7

Comment (MilapNaik) [reply package.json:59]: resolved by upgradign node to 14.17

File: src/RokuDeploy.spec.ts [hunk 2]
```diff
@@ -8,15 +8,14 @@ import * as path from 'path';
...
-import { RokuDeploy } from './RokuDeploy';
 import * as errors from './Errors';
 import { util, standardizePath as s, standardizePathPosix as sp } from './util';
 import type { FileEntry, RokuDeployOptions } from './RokuDeployOptions';
 import { cwd, expectPathExists, expectPathNotExists, expectThrowsAsync, outDir, rootDir, stagingDir, tempDir, writeFiles } from './testUtils.spec';
 import { createSandbox } from 'sinon';
 import * as r from 'postman-request';
 import type * as requestType from 'request';
-import type { CaptureScreenshotOptions, ConvertToSquashfsOptions, CreateSignedPackageOptions, DeleteDevChannelOptions, GetDevIdOptions, GetDeviceInfoOptions, RekeyDeviceOptions, SendKeyEventOptions, SideloadOptions } from './RokuDeploy';
+import { RokuDeploy, type CaptureScreenshotOptions, type ConvertToSquashfsOptions, type CreateSignedPackageOptions, type DeleteDevChannelOptions, type GetDevIdOptions, type GetDeviceInfoOptions, type RekeyDeviceOptions, type SendKeyEventOptions, type SideloadOptions } from './RokuDeploy';
```
Comment (TwitchBronBron) [inline src/RokuDeploy.spec.ts:18]: Let's revert this one. I prefer having 1 "import {notTypeStuff}" and then another line for "import type {typeStuff}". Freel free to tweak any eslint rules that complain about that.

File: src/RokuDeploy.ts [hunk 3]
```diff
@@ -470,7 +470,7 @@ export class RokuDeploy {
     }
 
     private checkRequest(results) {
-        if (!results || !results.response || typeof results.body !== 'string') {
+        if (!results?.response || typeof results?.body !== 'string') {
```
Comment (TwitchBronBron) [inline src/RokuDeploy.ts:473]: These optional chaining things cause issues with code coverage. Let's revert this and adjust any eslint rule that's complaining about it.

File: src/RokuDeploy.spec.ts (hunk 2 above)
Comment (MilapNaik) [reply src/RokuDeploy.spec.ts:18]: The only thing is this no duplicate import rule doesn't have an option to exclude this scenario and to disable it means it won't create an error if we import from the same file multiple times, which I think that is a good error to have right?

https://eslint.org/docs/latest/rules/no-duplicate-imports

File: src/RokuDeploy.ts (hunk 3 above)
Comment (MilapNaik) [reply src/RokuDeploy.ts]: done

File: src/RokuDeploy.spec.ts (hunk 2 above)
Comment (MilapNaik) [reply src/RokuDeploy.spec.ts]: I just disabled it for the one line

File: src/RokuDeploy.spec.ts [hunk 4]
```diff
@@ -8,14 +8,15 @@ import * as path from 'path';
...
-import { RokuDeploy } from './RokuDeploy';
 import * as errors from './Errors';
 import { util, standardizePath as s, standardizePathPosix as sp } from './util';
 import type { FileEntry, RokuDeployOptions } from './RokuDeployOptions';
 import { cwd, expectPathExists, expectPathNotExists, expectThrowsAsync, outDir, rootDir, stagingDir, tempDir, writeFiles } from './testUtils.spec';
 import { createSandbox } from 'sinon';
 import * as r from 'postman-request';
 import type * as requestType from 'request';
+import { RokuDeploy } from './RokuDeploy';
+// eslint-disable-next-line no-duplicate-imports
```
Comment (TwitchBronBron) [inline src/RokuDeploy.spec.ts:19]: Hmm. we shouldn't need this eslint disable comment. All of our other projects support 1 `import` and 1 `import type`, so not sure what's different about this project. 
(see this example from brighterscript)
image

**Review by TwitchBronBron — CHANGES_REQUESTED** [review]
This looks great. Just one issue with the eslint-disable comment, then this is good to go

File: src/RokuDeploy.spec.ts (hunk 4 above)
Comment (MilapNaik) [reply src/RokuDeploy.spec.ts]: Followed guidance from https://github.com/typescript-eslint/typescript-eslint/issues/2315 and used eslint plugin import

**Pull request merged**